src/modules/template-details/TemplateDetails.tsx

```diff
@@ -56,105 +59,129 @@ const TemplateDetails = ({ match }: TemplateDetails.Props) => {
     setConfirmDeleteOpen(false);
   }, []);
 
+  const forkTemplateHandler = async () => {
+    info.incrementForks();
+    setForkInProgress(true);
+    forkTemplate(match.params.id).then(
+      (response) => console.log("Response: " + response.data),
+      (reason) => console.log("Reason: " + reason)
+    ).then(
+      () => {
+        setForkInProgress(false);
+        info.completeFork(match.params.id, false);
+        info.decrementForks();
+      }
+
+    );
+  }
+
   const patternsNames = useMemo(toNames(template), [template]);
 
   return (
     <div className={csx.templateDetails}>
       {loading ? (
         <Loader />
       ) : (
-        !error && (
-          <>
-            {confirmDeleteOpen && (
-              <ConfirmDelete template={template} onClose={closeConfirmDelete} />
-            )}
-
-            <header>
-              <NavLink to={`${match.url}/documentation`}>
-                <Button>
-                  <MenuBookIcon /> DOCS
+          !error && (
+            <>
+              {confirmDeleteOpen && (
+                <ConfirmDelete template={template} onClose={closeConfirmDelete} />
+              )}
+
+              <header>
+                <NavLink to={`${match.url}/documentation`}>
+                  <Button>
+                    <MenuBookIcon /> DOCS
                 </Button>
-              </NavLink>
+                </NavLink>
+
+                <Button disabled={forkInProgress} onClick={forkTemplateHandler}>
 
-              <a href={template.githubLink} target="_blank">
-                <Button>
-                  <ShareIcon /> SOURCE
+                  <img src="https://www.iconsdb.com/icons/preview/white/fork-2-xxl.png"
+                    height="20"
+                    style={{ marginRight: "10px" }} />
+                  FORK
                 </Button>
-              </a>
 
```

Comment:
Avoid inline - styles.
In React we using inline styles only if we need dynamic logic or by convention. Don't try to mix this